File: Examples/test-suite/matlab/overload_extend_runme.m
import overload_extend.*

f = overload_extend.Foo();
if (f.test() ~= 0)
  error('Failed!')
end
if (f.test(3) ~= 1)
    error('Failed!')
end
if (f.test('hello') ~= 2)
    error('Failed!')
end
if (f.test(3,2) ~= 5)
    error('Failed!')
end
if (f.test(3.1) ~= 1003.1)
    error('Failed!')
end
if (f.test(3.0) ~= 1003)
    error('Failed to use "double" overload!')
end

File: Examples/test-suite/javascript/overload_extend_runme.js
var overload_extend = require("overload_extend")

var f = new overload_extend.Foo()
if (/* await */(f.test()) != 0) {
    throw new Error;
}
if (/* await */(f.test(3)) != 1) {
    throw new Error;
}
if (/* await */(f.test("hello")) != 2) {
    throw new Error;
}
if (/* await */(f.test(3, 2)) != 5) {
    throw new Error;
}
// In JavaScript there is no difference between 3.0 and 3
if (/* await */(f.test(3.0)) != 1) {
    throw new Error;
}
